src/PackageUrl.cs

```diff
@@ -263,8 +263,8 @@ private string ValidateNamespace(string @namespace)
             }
             return Type switch
             {
-                "vsm" or "cran" => WebUtility.UrlDecode(@namespace),
-                _ => WebUtility.UrlDecode(@namespace.ToLower())
+                "bitbucket" or "github" or "pypi" or "gitlab" => WebUtility.UrlDecode(@namespace.ToLower()),
```

Comment:
Looks like we don't have a test case for github and gitlab which exercises this branch.

Reply:
I added a new test case for github.  I did not see a good example for in the PURL types document (https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst), or in other test suite data files for gitlab, and I didn't want to add in a bad example that got copied by others.